x-pack/plugins/security_solution/server/endpoint/routes/actions/file_upload_handler.test.ts

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HttpApiTestSetupMock } from '../../mocks';
+import { createHttpApiTestSetupMock } from '../../mocks';
+import type { UploadActionRequestBody } from '../../../../common/endpoint/schema/actions';
+import type { getActionFileUploadHandler } from './file_upload_handler';
+import { registerActionFileUploadRoute } from './file_upload_handler';
+import { UPLOAD_ROUTE } from '../../../../common/endpoint/constants';
+import { getEndpointAuthzInitialStateMock } from '../../../../common/endpoint/service/authz/mocks';
+import { EndpointAuthorizationError } from '../../errors';
+import type { HapiReadableStream } from '../../../types';
+import { createHapiReadableStreamMock } from '../../services/actions/mocks';
+import {
+  createFile as _createFile,
+  deleteFile as _deleteFile,
+  setFileActionId as _setFileActionId,
+} from '../../services';
+import { EndpointActionGenerator } from '../../../../common/endpoint/data_generators/endpoint_action_generator';
+import { CustomHttpRequestError } from '../../../utils/custom_http_request_error';
+import type { ActionDetails } from '../../../../common/endpoint/types';
+import { omit } from 'lodash';
+
+jest.mock('../../services');
+const createFileMock = _createFile as jest.Mock;
+const deleteFileMock = _deleteFile as jest.Mock;
+const setFileActionIdMock = _setFileActionId as jest.Mock;
+
+describe('Upload response action create API handler', () => {
+  type UploadHttpApiTestSetupMock = HttpApiTestSetupMock<never, never, UploadActionRequestBody>;
+
+  let testSetup: UploadHttpApiTestSetupMock;
+  let httpRequestMock: ReturnType<UploadHttpApiTestSetupMock['createRequestMock']>;
+  let httpHandlerContextMock: UploadHttpApiTestSetupMock['httpHandlerContextMock'];
+  let httpResponseMock: UploadHttpApiTestSetupMock['httpResponseMock'];
+
+  beforeEach(() => {
+    testSetup = createHttpApiTestSetupMock<never, never, UploadActionRequestBody>();
+
+    ({ httpHandlerContextMock, httpResponseMock } = testSetup);
+    httpRequestMock = testSetup.createRequestMock();
+  });
+
+  describe('registerActionFileUploadRoute()', () => {
+    it('should register the route', () => {
+      registerActionFileUploadRoute(testSetup.routerMock, testSetup.endpointAppContextMock);
+
+      expect(testSetup.getRegisteredRouteHandler('post', UPLOAD_ROUTE)).toBeDefined();
+    });
+
+    it('should NOT register route if feature flag is false', () => {
+      // @ts-expect-error
```

Comment:
any specific error here?

Reply:
Yeah, assignment to a Readonly property. I'll have to adjust the test render mock at some point to perhaps not set these to `readonly` so that we don't have to do this.